File: app/src/main/java/com/example/minilist/main/MainPresenter.java
package com.example.minilist.main;

import java.io.File;

public class MainPresenter {

    public final static String FILE_NAME = "miniList";

    private final MainWireframe wireframe;

    public MainPresenter(MainWireframe wireframe) {
        this.wireframe = wireframe;
        checkFileExists();
    }

    public void checkFileExists() {
        if (new File(wireframe.getDir(), FILE_NAME).exists()) {
            wireframe.setExist();
        } else {
            wireframe.setNew();
        }
    }

    public void onEditButtonClicked() {
        wireframe.changeActivity(MainWireframe.Direction.EDIT);
    }

    public void onNewButtonClicked() {
        wireframe.changeActivity(MainWireframe.Direction.EDIT);
    }

    public void onShowButtonClicked() {
        wireframe.changeActivity(MainWireframe.Direction.WATCH);
    }

    public void onSettingsButtonClicked() {
        wireframe.changeActivity(MainWireframe.Direction.SETTINGS);
    }

}

File: app/src/main/java/com/example/minilist/main/MainPresenter.kt
package com.example.minilist.main

import com.example.minilist.Shared
import java.io.File

class MainPresenter(private val wireframe: MainWireframe) {

    init {
        checkFileExists()
    }

    fun checkFileExists() {
        if (File(wireframe.getFilesDir(), Shared.FILE_NAME).exists()) {
            wireframe.setExist()
        } else {
            wireframe.setNew()
        }
    }

    fun onEditButtonClicked() {
        wireframe.changeActivity(MainWireframe.Direction.EDIT)
    }

    fun onNewButtonClicked() {
        wireframe.changeActivity(MainWireframe.Direction.EDIT)
    }

    fun onShowButtonClicked() {
        wireframe.changeActivity(MainWireframe.Direction.WATCH)
    }

    fun onSettingsButtonClicked() {
        wireframe.changeActivity(MainWireframe.Direction.SETTINGS)
    }
}